src/assets/wise5/authoringTool/project-authoring-lesson/project-authoring-lesson.component.ts

```diff
@@ -43,4 +45,17 @@ export class ProjectAuthoringLessonComponent {
     this.expanded = !this.expanded;
     this.onExpandedChanged.emit({ id: this.lesson.id, expanded: this.expanded });
   }
+
+  protected delete(event: Event): void {
+    event.stopPropagation();
+    if (confirm($localize`Are you sure you want to delete this lesson?`)) {
+      this.deleteNodeService.deleteNode(this.lesson.id);
+      this.saveAndRefreshProject();
+    }
+  }
+
+  private saveAndRefreshProject(): void {
+    this.projectService.saveProject();
+    this.projectService.refreshProject();
+  }
```

Comment:
I'm wondering if extracting out saveAndRefreshProject() is a bit premature, since no other code in the class calls it atm? We can extract it out later when needed? 
```suggestion
  protected delete(event: Event): void {
    event.stopPropagation();
    if (confirm($localize`Are you sure you want to delete this lesson?`)) {
      this.deleteNodeService.deleteNode(this.lesson.id);
      this.projectService.saveProject();
      this.projectService.refreshProject();
    }
  }
```

Reply:
Yea there is some common code between ProjectAuthoringStepComponent and ProjectAuthoringLessonComponent that we can refactor later.